roseau/load_flow/units.py

```diff
@@ -41,10 +45,61 @@
 ureg.define("volt_ampere_reactive = 1 * volt_ampere = VAr")
 
 if TYPE_CHECKING:
-    Q_: TypeAlias = PlainQuantity[T]
+    # Copy types from pint and add complex
+    Scalar: TypeAlias = int | float | Decimal | Fraction | complex | np.number[Any]
+    Array: TypeAlias = np.ndarray[Any, Any]
+    UnitLike = str | dict[str, Scalar] | UnitsContainer | Unit
+
+    NpNumT = TypeVar("NpNumT", bound=np.number[Any])
+    MagBound = Scalar | Array | Sequence[Scalar | Array] | Sequence[Sequence[Scalar | Array]]
+    MagT = TypeVar("MagT", bound=MagBound)
+    MagT_co = TypeVar("MagT_co", covariant=True, bound=MagBound)
+
+    # Redefine Q_ with support for complex and better type hints
+    class Q_(NumpyQuantity[MagT_co]):  # type: ignore # noqa: N801
```

Comment:
This is done to add complex number support to the quantity class and to make the typing compatible with both VSCode and Pycharm (I tested both)